xbmc/addons/AddonManager.h

```diff
@@ -69,6 +69,12 @@ namespace ADDON
     bool Init();
     void DeInit();
 
+    CAddonMgr();
```

Comment:
I would keep this as simple as possible. Why trying to protect something that is very unlikely to happen? I can't think of any reason why s.o. would create another instance of CAddonMgr. If so I would be interested in that case :)

Copy and assignment constructors are only default that I would drop.